packages/react-components/react-carousel-preview/library/src/components/CarouselSlider/CarouselSlider.types.ts

```diff
@@ -0,0 +1,47 @@
+import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { MutableRefObject } from 'react';
+
+export type CarouselSliderSlots = {
+  /**
+   * The div that will be animated to track and center current index
+   */
+  root: Slot<'div'>;
+  /**
+   * The viewport/window of the carousel
+   */
+  container: NonNullable<Slot<'div'>>;
+};
+
+/**
+ * CarouselSlider Props
+ */
+export type CarouselSliderProps = Partial<ComponentProps<CarouselSliderSlots>> & {
```

Comment:
Consider adding props to toggle between vertical and horizontal?

Reply:
A vertical carousel would be interesting, we haven't seen this spec'd but can definitly add if we have consumer interest.